### ISSUE TITLE
Unknown argument type when multiple generics passed to function

### ISSUE THREAD

**Issue by mehdigmira** [microsoft/pyright#5804]
**Describe the bug**
Following snippet causes an error with latest version of pyright. Note that If `collect` takes only one input arg, no error is logged.

**Code or Screenshots**

```python
# pyright: strict

from __future__ import annotations

from typing import Generic, TypeVar

_T = TypeVar("_T")


class A:
    ...


class B(A, Generic[_T]):
    def __init__(self, a: _T) -> None:
        ...


def collect(arg1: A, arg2: A):
    ...


# this works fines
b1, b2 = B(1), B(2)
collect(b1, b2)

# but this logs an error
# - error: Argument type is partially unknown
#     Argument corresponds to parameter "args" in function "collect"
#     Argument type is "B[Unknown]" (reportUnknownArgumentType)
# 1 error, 0 warnings, 0 informations
collect(B(1), B(2))

```

**Comment by erictraut** [comment]
Thanks for the bug report.

This was a regression that I introduced when fixing another recent bug you reported (#5693). Both cases are now covered by unit tests, so that prevent similar regressions in the future.

**Comment by mehdigmira** [comment]
yeah, thought it would be related, it had a familiar look & feel :)

**Comment by erictraut** [comment]
This is included in pyright 1.1.325, which I just published. It will also be included in an upcoming pylance release.
